overlord/ifacestate/apparmorprompting/prompting_test.go

```diff
@@ -217,7 +225,12 @@ func (s *apparmorpromptingSuite) TestHandleListenerErrors(c *C) {
 	}
 	reqChan <- req
 	time.Sleep(10 * time.Millisecond)
-	c.Check(fmt.Errorf("%#v", logbuf.String()), ErrorMatches, ".*Error while handling request: cannot get abstract permissions from empty AppArmor permissions.*")
+	logger.WithLoggerLock(func() {
+		c.Check(logbuf.String(), testutil.Contains,
+			" WARNING: too many outstanding prompts for user 1000; auto-denying new one\n")
+	})
+
+	c.Assert(mgr.Stop(), IsNil)
```

Comment:
shouldn't we defer these as early as reasonable instead?

Reply:
When these are deferred the tests explode due to the way the listener is mocked, unfortunately